internal/clients/diffreviewer/circleci/circleci_service.go

```diff
@@ -0,0 +1,106 @@
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/nightfallai/nightfall_code_scanner/internal/clients/diffreviewer"
+	"github.com/nightfallai/nightfall_code_scanner/internal/clients/gitdiff"
+	"github.com/nightfallai/nightfall_code_scanner/internal/clients/logger"
+	circlelogger "github.com/nightfallai/nightfall_code_scanner/internal/clients/logger/circle_logger"
+	"github.com/nightfallai/nightfall_code_scanner/internal/clients/nightfall"
+	"github.com/nightfallai/nightfall_code_scanner/internal/interfaces/gitdiffintf"
+	"github.com/nightfallai/nightfall_code_scanner/internal/nightfallconfig"
+)
+
+const (
+	WorkspacePathEnvVar          = "GITHUB_WORKSPACE"
+	NightfallAPIKeyEnvVar        = "NIGHTFALL_API_KEY"
+	CircleRepoNameEnvVar         = "CIRCLE_PROJECT_REPONAME"
+	CircleOwnerNameEnvVar        = "CIRCLE_PROJECT_USERNAME"
+	CircleCurrentCommitShaEnvVar = "CIRCLE_SHA1"
+	CircleBeforeCommitEnvVar     = "EVENT_BEFORE"
+	CirclePullRequestUrlEnvVar   = "CIRCLE_PULL_REQUEST"
+)
+
+// Service contains the github client that makes Github api calls
+type Service struct {
+	Logger  logger.Logger
+	GitDiff gitdiffintf.GitDiff
+}
+
+// NewCircleCiService creates a new CircleCi service
+func NewCircleCiService() diffreviewer.DiffReviewer {
+	return &Service{
+		Logger: circlelogger.NewDefaultCircleLogger(),
+	}
+}
+
+// GetLogger gets the github service logger
+func (s *Service) GetLogger() logger.Logger {
+	return s.Logger
+}
+
+// LoadConfig gets all config values from files or environment and creates a config
+func (s *Service) LoadConfig(nightfallConfigFileName string) (*nightfallconfig.Config, error) {
+	s.Logger.Debug("Loading configuration")
+	workspacePath, ok := os.LookupEnv(WorkspacePathEnvVar)
+	if !ok {
+		s.Logger.Error(fmt.Sprintf("Environment variable %s cannot be found", WorkspacePathEnvVar))
+		return nil, errors.New("missing env var for workspace path")
+	}
+	commitSha, ok := os.LookupEnv(CircleCurrentCommitShaEnvVar)
+	if !ok {
+		s.Logger.Error(fmt.Sprintf("Environment variable %s cannot be found", CircleCurrentCommitShaEnvVar))
+		return nil, errors.New("missing env var for commit sha")
+	}
+	beforeCommitSha, ok := os.LookupEnv(CircleBeforeCommitEnvVar)
+	if !ok {
+		s.Logger.Error(fmt.Sprintf("Environment variable %s cannot be found", CircleBeforeCommitEnvVar))
+		return nil, errors.New("missing env var for prev commit sha")
+	}
+	s.GitDiff = &gitdiff.GitDiff{
+		WorkDir:    workspacePath,
+		BaseBranch: "master", //TODO: look into how to get this instead of hardcoding
+		BaseSHA:    beforeCommitSha,
+		Head:       commitSha,
+	}
+	nightfallConfig, err := nightfallconfig.GetNightfallConfigFile(workspacePath, nightfallConfigFileName)
+	if err != nil {
+		s.Logger.Error("Error getting Nightfall config file. Ensure you have a Nightfall config file located in the root of your repository at .nightfalldlp/config.json with at least one Detector enabled")
+		return nil, err
+	}
+	nightfallAPIKey, ok := os.LookupEnv(NightfallAPIKeyEnvVar)
+	if !ok {
```

Comment:
probably need to add an extra `!ok && nightfallAPIKey != ""`, otherwise if they pass in a NIGHTFALL_API_KEY env var to their configuration but forget to actually set that value in their Circle secrets, this will pass